lib/kinesis_client/stream/coordinator.ex

```diff
@@ -116,12 +116,21 @@ defmodule KinesisClient.Stream.Coordinator do
 
         {:noreply, %{state | shard_graph: shard_graph, shard_ref_map: map}}
 
-      other ->
-        Logger.info("Stream is not in active state, sleeping... Stream state: #{inspect(other)}")
+      {:error, reason} ->
+        Logger.info("[kcl_ex] error describing stream shards with result #{inspect(reason)}")
+
+        if state.startup_attempt < @max_startup_attempts do
+          :timer.sleep(state.retry_timeout)
+          state = %{state | startup_attempt: state.startup_attempt + 1}
+          notify({:retrying_describe_stream, self()}, state)
+          {:noreply, state, {:continue, :describe_stream}}
+        else
+          Logger.error(
+            "[kcl_ex] error starting stream coordinator after three attempts for stream #{state.stream_name}"
+          )
 
-        :timer.sleep(state.retry_timeout)
-        notify({:retrying_describe_stream, self()}, state)
-        {:noreply, state, {:continue, :describe_stream}}
+          {:stop, :normal, state}
```

Comment:
what happens to the rest of the application if this GenServer stops?

Reply:
The entire kcl_ex supervision tree will shutdown in this case, and it will be captured by our circuit breaker monitor as a crash and we will get alerted as expected.

Reply:
Well, actually, kcl_ex will restart itself two more times as part of its supervision logic, and go thru the motions again. But, if all three attempts fail and end with a manual stop, we'll get a proper crash captured by the circuit breaker.

This is essentially equivalent to what happens right now without any backoff handling of rate limits that I've added here.